src/Rule/NumberHandler.php

```diff
@@ -32,7 +32,7 @@ public function validate(mixed $value, object $rule, ValidationContext $context)
         if (is_bool($value) || !is_scalar($value)) {
             $result->addError($rule->getNotANumberMessage(), [
                 'attribute' => $context->getAttribute(),
-                'value' => $value,
+                'valueType' => get_debug_type($value),
```

Comment:
For same message must be fixed attributes list.

Reply:
https://github.com/yiisoft/validator/pull/364#discussion_r1013869773